skopt/optimizer/optimizer.py

```diff
@@ -152,6 +155,10 @@ def __init__(self, dimensions, base_estimator,
 
         self.n_jobs = n_jobs
 
+        # the cache of responses of ask. For n_points not None, this ensures
+        # that multiple calls to ask return same sets of points.
+        self.cache = {}
```

Comment:
Note that the behaviour between `n_points=None` and `n_points=x` is different when `n_random_starts` is not exhausted and `ask` is called multiple times. Also can you make it `self.cache_`